### PR TITLE
Add pyproject.toml to Python.gitignore

### PR DESCRIPTION
**Pull request by geoqiao**
# Pull Request

Thank you for contributing to @toptal/gitignore and https://www.gitignore.io.

## New or update

Select the appropriate check box for this pull request. This helps when merging to ensure there are no conflicts with other templates or misunderstandings of how thee template list works.

### New

- [ ] Template - New `.gitignore` template
- [ ] Composition - Template made from smaller templates
- [ ] Inheritance - Template similar to an existing template
- [ ] Patch - Template extending functionality of existing template

### Update

- [X] Template - Update existing `.gitignore` template

## Details

*Add pyproject.toml to Python.gitignore*

### CODE REVIEW

**Comment by baggiponte** [comment]
heya! as a pythonista, this feels weird. `pyproject.toml` is not meant to be in `.gitignore`. it's like `Cargo.toml`.

File: templates/Python.gitignore [hunk 1]
```diff
@@ -109,6 +109,10 @@ ipython_config.py
 #   https://pdm.fming.dev/#use-with-ide
 .pdm.toml
 
+# rye
+#   rye stores project-wide configurations in pyproject.toml, but it is recommended to not include it in version control
```
Comment (edgarrmondragon) [inline templates/Python.gitignore:113]: I don't think rye recommends to ignore the file, and even if that was true, other tools and PEP 517 require this file to be present in version control. cc @mitsuhiko.

Comment (mitsuhiko) [reply templates/Python.gitignore:113]: Yeah rye does not recommend against committing this file. Hope we don't have bad docs somewhere.

Comment (geoqiao) [reply templates/Python.gitignore:113]: Yes, I just realized the issue you mentioned.
When I submitted this PR, I simply didn't want my personal email to be uploaded to GitHub.
It was a special case for me and shouldn't be considered as the default option for the `.gitignore` file.
Thank you very much for bringing this to my attention!

**Pull request closed without merging**